src/python/pants/binaries/binary_util.py

```diff
@@ -103,8 +103,9 @@ def _select_binary_base_path(self, supportdir, version, name, uname_func=None):
     try:
       middle_path = self._path_by_id[os_id]
     except KeyError:
-      raise self.MissingMachineInfo('Update --binaries-path-by-id to find binaries for {!r}'
-                                    .format(os_id))
+      raise self.MissingMachineInfo(('Unable to find binary {name} version {version}. ' +
+                                    'Update --binaries-path-by-id to find binaries for {os_id!r}')
+                                    .format(name=name, version=version, os_id=os_id))
```

Comment:
thanks to python's implicit string concatenation, you don't need the `+` or extra `()`'s here:

```
>>> ('this is '
...  'an implicitly concatenated string '
...  'that can be formatted: {}'
...  .format('see'))
'this is an implicitly concatenated string that can be formatted: see'
```

Reply:
I really dislike implicit string concatenation :(

Done.